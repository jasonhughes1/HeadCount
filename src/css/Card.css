.card-component {
  background-color: white;
  box-shadow: 0px 0px 20px 0px rgba(0, 0, 0, 0.1);
  margin: 6px;
  width: 251px;
  padding: 5px;
  border-radius: 5px;
  cursor: pointer; }

.card-component:hover {
  background-color: #eff2fa; }

.card-info-block {
  width: 116px;
  margin: 0 auto; }

.info-span {
  font-weight: 900;
  margin-right: 10px; }

hr {
  border: 2px solid black;
  opacity: .06;
  width: 90%; }

.change-green {
  color: #0d0262;
  opacity: .7; }

.highlighted {
  background-color: #f9ead9; }

.highlighted:hover {
  background-color: #f6ddc1; }
