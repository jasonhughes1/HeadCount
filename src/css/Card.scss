.card-component {
	background-color: white;
	box-shadow: 0px 0px 20px 0px rgba(0,0,0,0.1);
	margin: 6px;
	width: 251px;
	padding: 5px;
	border-radius: 5px;
	cursor: pointer;

	
}

.card-component:hover {
	background-color: rgba(255,255,255,.7);
}

.card-info-block {
	width: 116px;
	margin: 0 auto;
}

.info-span {
	font-weight: 900;
	margin-right: 10px;
}

hr {
	border: 2px solid #2c89bf;
	opacity: .16;
	width: 90%;
}

.change-green {
	color: #0e5d8c;
  opacity: .7;
}

.highlighted {
	background-color: rgba(255,255,255,.7);
}

.highlighted:hover {
	background-color: rgba(255,255,255,.55);
}

