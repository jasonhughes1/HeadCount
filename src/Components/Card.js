import React from 'react';
import '../css/Card.css';

// let colorDisplay = () => {
//   if (Number(data[year]) > .5) {
//     return ( <span className="info-span">{year}:{data[year]} </span> )
//   } else {
//     return ( <span className="info-span color-green">{year}:{data[year]} </span> )
//   }


const Card = ({ location, data, id, updateCardToCompare }) => {
  let yearArray = Object.keys(data);
  let mappedYearArray = yearArray.map((year, index)=> {

  return  <p key={`${id}-${index}`}>
            <span className="info-span color-green">{year}:</span>{data[year]} 
          </p>
  })

  return (
    <div onClick={() => updateCardToCompare(location)}
       className="card-component">

      <h2>{location}</h2>
      <hr />
      <div className="card-info-block">
        {mappedYearArray}
      </div>
    </div>
  )
}

export default Card;
