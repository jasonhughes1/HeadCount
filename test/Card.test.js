import React from 'react'
import { shallow } from 'enzyme'
import Card from './../src/Components/Card';

describe('Card Tests', () => {
  it('should match the snapshot', () => {
    const card = { location: 'Turing', data: 1 }
    const { location, data } = card
    const renderedApp = shallow(<Card
                            location={location}
                                data={data}  />)

   expect(renderedApp).toMatchSnapshot()
  })
})
