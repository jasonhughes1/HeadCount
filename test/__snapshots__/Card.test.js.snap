exports[`Card Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={1}
    location="Turing" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Turing
</h2>,
        <hr />,
        <div
          className="card-info-block">
          
</div>,
      ],
      "className": "card-component",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Turing",
        },
        "ref": null,
        "rendered": "Turing",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "card-info-block",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Turing
</h2>,
          <hr />,
          <div
            className="card-info-block">
            
</div>,
        ],
        "className": "card-component",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "card-info-block",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
